### ISSUE TITLE
v0.2.6 completely breaks Prettier in Svelte Projects

### ISSUE THREAD

**Issue by n00ki** [tailwindlabs/prettier-plugin-tailwindcss#148]
Node version: `18.12.1`
Package Manager: `pnpm 8.2.0`
VSCode version: `1.77.1`
`prettier` version: `2.8.7`
`prettier-plugin-tailwindcss` version: `0.2.6`
`tailwindcss`version: `3.3.1`
OS: `macOS Ventura 13.3`

Hi :)
I'm dealing with a significant performance issue across all of my SvelteKit projects, after upgrading `prettier-plugin-tailwindcss` to `v0.2.6`.
Upon launch, CPU climbs to 100% and higher, which makes coding basically impossible. it seems like the app shell is doing fine, yet the code editing experience (auto-completion, formatting, saving, etc.) completely breaks.
After running the Bisect tool, as the CPU hogging process is `Code Helper (Plugin)` (or `ExtensionHost`), It seems like Prettier is the root cause.

It's important to mention that I didn't make any changes in VSCode/prettier configurations.
removing `prettier-plugin-tailwindcss` entirely from the prettier plugins array / downgrading it to 0.2.5 fixed the issue immediately.

I've attached some screenshots of the activity monitor, VSCode Process Explorer, ps aux, and messages displayed when attempting to save Svelte / TS files (while saving is prevented, and vscode is basically stuck).

<img width="1250" alt="Activity Monitor" src="https://user-images.githubusercontent.com/42724748/230841752-aeb2bdb8-6f42-4962-bdc0-322fe65feb92.png">
<img width="1250" alt="Activity Monitor 2" src="https://user-images.githubusercontent.com/42724748/230841759-2e7a7891-bf1e-46ae-8956-78d02ea2db2a.png">
<img width="1770" alt="Extension Process" src="https://user-images.githubusercontent.com/42724748/230841765-e8e16784-1083-4f8d-9cf5-db266e031671.png">
<img width="1714" alt="Process Explorer" src="https://user-images.githubusercontent.com/42724748/230841771-50eee709-0585-404b-9a4d-408060eed18a.png">
<img width="456" alt="Saving Svelte Files" src="https://user-images.githubusercontent.com/42724748/230841775-f937d8a0-7aa7-4435-be70-ec31bd921785.png">
<img width="456" alt="Saving TS Files" src="https://user-images.githubusercontent.com/42724748/230841776-c59eea68-7d88-49d4-9c11-2ffae3b29ff1.png">

```
// .prettierrc
{
  "useTabs": false,
  "singleQuote": true,
  "trailingComma": "none",
  "printWidth": 120,
  "bracketSameLine": true,
  "plugins": ["prettier-plugin-svelte", "prettier-plugin-tailwindcss"],
  "pluginSearchDirs": false,
  "overrides": [
    {
      "files": "*.svelte",
      "options": {
        "parser": "svelte",
        "svelteSortOrder": "scripts-markup-styles",
        "svelteBracketNewLine": false,
        "svelteAllowShorthand": true,
        "svelteIndentScriptAndStyle": true
      }
    }
  ]
}
```

Any ideas? 💡

**Comment by thecrypticace** [comment]
Hey! This will likely be impossible to debug without a reproduction project to test in. This does not happen to me in a freshly set up SvelteKit project and I tried best replicating your set up (node, pnpm, prettier version, Ventura, etc…). Can you create a new project and let me know what you see there? Also is there any project/repo this happens in that you can share? Even a stripped down version that still produces problems?

It would also be beneficial to have a list of your installed VS Code plugins and settings relating to formatting. I always have trouble with pnpm and the prettier extension not working **at all** unless I use "Format Code (Forced)" and I'm not sure if that's what you're doing here or not.

I suspect it has something to do with the new config resolution for TypeScript and ESM configs but I can't figure out why that would be a problem. If it's using ~90–100% of the CPU that makes it sound like it's stuck in a loop for some reason.

**Comment by thecrypticace** [comment]
Basically, what I want to be able to do is reproduce this locally and then profile the process in Instruments and tracking syscalls using `dtruss` (have to reboot and enable it because of SIP :/) to see what could be happening here.

That'll at least point me in a direction as to what might be going on.

**Comment by n00ki** [comment]
> Hey! This will likely be impossible to debug without a reproduction project to test in. This does not happen to me in a freshly set up SvelteKit project and I tried best replicating your set up (node, pnpm, prettier version, Ventura, etc…). Can you create a new project and let me know what you see there? Also is there any project/repo this happens in that you can share? Even a stripped down version that still produces problems?
> 
> 
> 
> It would also be beneficial to have a list of your installed VS Code plugins and settings relating to formatting. I always have trouble with pnpm and the prettier extension not working **at all** unless I use "Format Code (Forced)" and I'm not sure if that's what you're doing here or not.
> 
> 
> 
> I suspect it has something to do with the new config resolution for TypeScript and ESM configs but I can't figure out why that would be a problem. If it's using ~90–100% of the CPU that makes it sound like it's stuck in a loop for some reason.

Hi 😀 
Thanks for reaching out!
The best I can do arm is:

https://github.com/n00ki/kickoff-sveltekit

Which is a starter template I made. The issue is also reproduced when working on it.

Hope it's good enough🙏🏼

**Comment by thecrypticace** [comment]
That's helpful! I left the project open in a Svelte file for almost an hour and merely typing in the editor makes vS Code stutter now. Not even related to saving the files which is quite odd. I'll see if I can figure out what's going on.

**Comment by stanf0rd** [comment]
Same for 0.2.7 versions — Svelte prettifying is really slow

**Comment by n00ki** [comment]
@thecrypticace any chance you were able to find the root cause for this behavior?

**Comment by andy1li** [comment]
Thanks for the issue! 

Downgrade back to 0.2.5 temporarily solved the extreme slow prettifying in my svelte project.

**Comment by thecrypticace** [comment]
@n00ki Not yet. The slowness I ended up experiencing appeared to have been a weird interaction with the Prettier plugin, an invalid config file, and copilot. As soon as I disabled co-pilot the typing speed returned to normal. So I think it's unrelated.

**Comment by SkyleLai** [comment]
I faced this problem too(Nuxt 3 instead of Svelte). I remove VS Code plugin "Prettier - Code formatter" and install "Prettier ESLint" seems to solve the problem.

**Comment by reinink** [comment]
Hey just a note on this — we were able to reproduce this issue and were able to verify that this occurs with the VS Code Prettier extension with nothing else enabled (using Extension Bisect).

Disabling the `prettier-plugin-tailwindcss` resolves the issue, so somehow the VS Code Prettier extension and this plugin are not getting along.

At this point we need to do some more debugging to see if we can figure out what's causing this.

**Comment by thecrypticace** [comment]
Hey! So we think we've got a fix for this. I've merged this in #153 and it's available via to test via our insiders build: `npm install prettier-plugin-tailwindcss@insiders`

Could you give it a test and report back? You'll need to close/reopen VS Code after installing it so the prettier extension doesn't have the old version in memory.

**Comment by n00ki** [comment]
> Hey! So we think we've got a fix for this. I've merged this in #153 and it's available to test via our insiders build:
> 
> ```
> npm install prettier-plugin-tailwindcss@insiders
> ```
> 
> Could you give it a test and report back? You'll need to close/reopen VS Code after installing it so the prettier extension doesn't have the old version in memory.

Gave it a go, looks great! seems like you did it :)
